api/config_entry.go

```diff
@@ -218,21 +218,22 @@ type UpstreamLimits struct {
 }
 
 type ServiceConfigEntry struct {
-	Kind             string
-	Name             string
-	Partition        string                  `json:",omitempty"`
-	Namespace        string                  `json:",omitempty"`
-	Protocol         string                  `json:",omitempty"`
-	Mode             ProxyMode               `json:",omitempty"`
-	TransparentProxy *TransparentProxyConfig `json:",omitempty" alias:"transparent_proxy"`
-	MeshGateway      MeshGatewayConfig       `json:",omitempty" alias:"mesh_gateway"`
-	Expose           ExposeConfig            `json:",omitempty"`
-	ExternalSNI      string                  `json:",omitempty" alias:"external_sni"`
-	UpstreamConfig   *UpstreamConfiguration  `json:",omitempty" alias:"upstream_config"`
-	Destination      *DestinationConfig      `json:",omitempty"`
-	Meta             map[string]string       `json:",omitempty"`
-	CreateIndex      uint64
-	ModifyIndex      uint64
+	Kind                  string
+	Name                  string
+	Partition             string                  `json:",omitempty"`
+	Namespace             string                  `json:",omitempty"`
+	Protocol              string                  `json:",omitempty"`
+	Mode                  ProxyMode               `json:",omitempty"`
+	TransparentProxy      *TransparentProxyConfig `json:",omitempty" alias:"transparent_proxy"`
+	MeshGateway           MeshGatewayConfig       `json:",omitempty" alias:"mesh_gateway"`
+	Expose                ExposeConfig            `json:",omitempty"`
+	ExternalSNI           string                  `json:",omitempty" alias:"external_sni"`
+	UpstreamConfig        *UpstreamConfiguration  `json:",omitempty" alias:"upstream_config"`
+	Destination           *DestinationConfig      `json:",omitempty"`
+	MaxInboundConnections int
```

Comment:
Should this use `json:",omitempty" alias:"max_inbound_connections"`

Reply:
The `alias` always confuses me because AFAIK Go can marshal to snake_case fine (there are passing unit tests for snake case). So I wonder if the aliases are actually perpetuating a misunderstanding of Go's marshalling magic

Reply:
I think we can keep the annotation here to make it consistent with the other fields. If we decide to remove them, we can do it altogether.